src/olympia/addons/tasks.py

```diff
@@ -653,9 +653,10 @@ def delete_addon_not_compatible_with_firefoxes(ids, **kw):
         ids[0], ids[-1], len(ids))
     qs = Addon.objects.filter(id__in=ids)
     for addon in qs:
-        addon.appsupport_set.filter(
-            app__in=(amo.THUNDERBIRD.id, amo.SEAMONKEY.id)).delete()
-        addon.delete()
+        with transaction.atomic():
+            addon.appsupport_set.filter(
+                app__in=(amo.THUNDERBIRD.id, amo.SEAMONKEY.id)).delete()
+            addon.delete()
```

Comment:
> This still breaks the whole task in case of an exception. I'm missing a try/except here that catches and logs separate exceptions in case they happen.

It's true, but at least it will be in a consistent state and we can just fire the task again once we fixed the issue.

> Also, this should actually be good to unittest by mocking the delete() of an add-on and making sure that assuport_set is still complete.

Transaction-aware tests are so slow...